packages/kotti-ui/source/kotti-field-file-upload/types.ts

```diff
@@ -0,0 +1,264 @@
+import { yocoIconSchema } from '@3yourmind/yoco'
+import { Yoco } from '@3yourmind/yoco'
+import { v4 as uuidv4 } from 'uuid'
+import { z } from 'zod'
+
+import { KottiField } from '../kotti-field/types'
+
+export namespace KottiFieldFileUpload {
+	export namespace Status {
+		export enum Common {
+			INVALID = 'INVALID',
+			NOT_STARTED = 'NOT_STARTED',
+			UPLOADED = 'UPLOADED',
+			UPLOADED_WITH_ERROR = 'UPLOADED_WITH_ERROR',
+		}
+		export const commonSchema = z.nativeEnum(Common)
+
+		export enum NonRemote {
+			READY_TO_UPLOAD = 'READY_TO_UPLOAD',
+		}
+		export const nonRemoteSchema = z.nativeEnum(NonRemote)
+
+		export enum Remote {
+			CANCELED = 'CANCELED',
+			ERROR = 'ERROR',
+			UPLOADING = 'UPLOADING',
+		}
+		export const remoteSchema = z.nativeEnum(Remote)
+
+		export const schema = commonSchema.or(nonRemoteSchema).or(remoteSchema)
+		export type All = z.output<typeof schema>
+	}
+
+	export enum Validation {
+		INVALID_EXTENSION = 'INVALID_EXTENSION',
+		MAX_SIZE_EXCEEDED = 'MAX_SIZE_EXCEEDED',
+		SUCCESS = 'SUCCESS',
+	}
+	export const validationSchema = z.nativeEnum(Validation)
+
+	export const valueSchema = z
+		.object({
+			file: z.custom<File>(),
+			id: z.string().uuid(),
+			status: Status.schema,
+		})
+		.or(
+			z.object({
+				downloadUrl: z.string().optional(),
+				id: z
+					.string()
+					.uuid()
+					.optional()
+					.default(() => uuidv4()),
```

Comment:
remember how we saw in Vue that the default doesn't work on that level :) 

because it can only default on a prop level

Reply:
id is no longer optional